src/puppeteer.js

```diff
@@ -0,0 +1,38 @@
+import puppeteer from 'puppeteer';
+import { checkParamOrThrow } from 'apify-client/build/utils';
+import { parseUrl } from './utils';
+import { ENV_VARS } from './constants';
+
+export const getDefaultPuppeteerOptions = () => ({
+    args: ['--no-sandbox'],
+    headless: !!process.env[ENV_VARS.HEADLESS],
+    proxyUrl: null,
+});
+
+/**
+ * @memberof module:Apify
+ * @function
+ * @description Launches Puppeteer preconfigured to work with Apify platform.
+ * The result of the function is a result of puppeteer.launch().
+ * @param options Optional settings, their defaults are provided in the getDefaultPuppeteerOptions function.
+ * @return Returns a promise.
+ */
+export const launchPuppeteer = (opts = {}) => {
+    checkParamOrThrow(opts, 'opts', 'Object');
+
+    const { proxyUrl } = opts;
+
+    if (proxyUrl) {
+        checkParamOrThrow(proxyUrl, 'opts.proxyUrl', 'String');
+
+        const { host, port, protocol } = parseUrl(proxyUrl);
+        if (!host || !port) throw new Error('Invalid "proxyUrl" option: the URL must contain hostname and port number.');
```

Comment:
I think we should also throw an error if password is specified, to make it more user-friendly. Otherwise people will not know what's going on.